platform/ios/src/MGLMapView.mm

```diff
@@ -5810,6 +5815,15 @@ - (void)didUpdateLocationWithUserTrackingAnimated:(BOOL)animated completionHandl
         return;
     }
 
+    CLLocationDirection direction = self.directionByFollowingWithCourse;
+    if (heading != nil && heading.headingAccuracy > 0) {
+        self.userLocation.heading = heading;
+        if (self.userTrackingMode == MGLUserTrackingModeFollowWithHeading)
+        {
+            direction = (heading.trueHeading >= 0 ? heading.trueHeading : heading.magneticHeading);
+        }
+    }
+
```

Comment:
why do we have this logic in this `didUpdateLocationWithUserTrackingAnimated:` method? it seems we only need this in `- (void)locationManager:(__unused id<MGLLocationManager>)manager didUpdateLocations:(NSArray *)locations heading:(nullable CLHeading *)heading animated:(BOOL)animated completionHandler:(nullable void (^)(void))completion` method, so we do this logic directly in that method, similar to what is currently available in `- (void)locationManager:(__unused id<MGLLocationManager>)manager didUpdateHeading:(CLHeading *)newHeading` method.

Reply:
Also, it seems like we simply ignored a few checks that exist in `locationManager:didUpdateHeading:` method. can you explain the need to ignore these checks?

Reply:
`updateUserLocationAnnotationView` `self.delegate mapView:self didUpdateUserLocation:self.userLocation` and `mbglMap.easeTo` methods are already called in `didUpdateLocations` method. So the only thing what is left was `direction`.